File: algorithms/1431.kids-with-the-greatest-number-of-candies/solution.go
func kidsWithCandies(candies []int, extraCandies int) []bool {
	maxCandies := 0
	res := make([]bool, len(candies))

	for _, v := range candies {
		if v > maxCandies {
			maxCandies = v
		}
	}

	for i, v := range candies {
		if v+extraCandies >= maxCandies {
			res[i] = true
		}
	}

	return res
}

File: algorithms/1431.kids-with-the-greatest-number-of-candies/solution.rb
# @param {Integer[]} candies
# @param {Integer} extra_candies
# @return {Boolean[]}
def kids_with_candies(candies, extra_candies)
  max_candies = candies.sort[-1]
  candies.map { |candy| candy + extra_candies >= max_candies }
end
